libafl_derive/src/lib.rs

```diff
@@ -69,3 +69,61 @@ pub fn libafl_serdeany_derive(input: TokenStream) -> TokenStream {
         libafl_bolts::impl_serdeany!(#name);
     })
 }
+
+/// Derive macro to implement `Display` for a struct where all fields implement `Display`.
+/// The result is the space separated concatenation of all fields' display.
+/// Order of declaration is preserved.
+/// Specifically handled cases:
+/// Options: Some => inner type display None => "".
+/// Vec: inner type display space separated concatenation.
+/// Generics and other more or less exotic stuff are not supported.
+#[proc_macro_derive(Display)]
+pub fn libafl_display(input: TokenStream) -> TokenStream {
```

Comment:
Is this used anywhere?

Reply:
Yep here: https://github.com/AFLplusplus/LibAFL/blob/4f174d5d3c3b91521f56a3b6d698a1f255d8afd1/libafl_qemu/src/qemu/qemu_config.rs#L267-L269

It is used to avoid the ugly error prone Display as pointed out by rmalmain. In this way if a field without `Display` is added to `QemuConfig`, compilation will fail.

Reply:
Want to add a simple example `Display` output to the docs so other ppl can decide to use it (or not use it)